jaeger-core/src/main/java/com/uber/jaeger/propagation/TextMapCodec.java

```diff
@@ -74,20 +76,30 @@ public SpanContext extract(TextMap carrier) {
       } else if (key.equals(Constants.DEBUG_ID_HEADER_KEY)) {
         debugId = decodedValue(entry.getValue());
       } else if (key.startsWith(baggagePrefix)) {
-        if (baggage == null) {
-          baggage = new HashMap<String, String>();
-        }
         baggage.put(keys.unprefixedKey(key, baggagePrefix), decodedValue(entry.getValue()));
+      } else if (key.equals(DEBUG_JAEGER_SPAN_BAGGAGE_HEADER)) {
+        /**
+         * debug jaeger baggage is loaded for a header item that looks like,
+         * <pre>
+         * {@code
+         *    "jaeger-baggage": "k1=v1,k2=v2"
+         * }
+         * </pre>
+         */
+        String decodedDebugBaggageItems = decodedValue(entry.getValue());
+        String[] debugBaggageItems = decodedDebugBaggageItems.split(",");
+        for (String debugBaggageItem: debugBaggageItems) {
+          String[] debugBaggageItemParts = debugBaggageItem.split("=");
+          baggage.put(debugBaggageItemParts[0], debugBaggageItemParts[1]);
```

Comment:
must check for malformed headers, current code can cause IndexOutOfBounds exception